File: sonet-web/src/types/workspace.ts
export interface Workspace {
  id: number
  name: string
  slug: string
  description: string | null
  owner_id: number
  is_personal: boolean
  icon: string | null
  is_encrypted: boolean
  created_at: string
  updated_at: string
}

export interface WorkspaceMember {
  id: number
  workspace_id: number
  user_id: number
  role: 'viewer' | 'editor' | 'admin'
  invited_by: number | null
  encrypted_dek?: string | null
  joined_at: string
}

File: sonet-api/internal/model/workspace.go
package model

import "time"

type Workspace struct {
	ID          uint64    `json:"id" db:"id"`
	Name        string    `json:"name" db:"name"`
	Slug        string    `json:"slug" db:"slug"`
	Description *string   `json:"description" db:"description"`
	OwnerID     uint64    `json:"owner_id" db:"owner_id"`
	IsPersonal  bool      `json:"is_personal" db:"is_personal"`
	Icon        *string   `json:"icon" db:"icon"`
	IsEncrypted bool      `json:"is_encrypted" db:"is_encrypted"`
	CreatedAt   time.Time `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time `json:"updated_at" db:"updated_at"`
}

type WorkspaceMember struct {
	ID           uint64    `json:"id" db:"id"`
	WorkspaceID  uint64    `json:"workspace_id" db:"workspace_id"`
	UserID       uint64    `json:"user_id" db:"user_id"`
	Role         string    `json:"role" db:"role"`
	InvitedBy    *uint64   `json:"invited_by" db:"invited_by"`
	EncryptedDEK *string   `json:"encrypted_dek,omitempty" db:"encrypted_dek"`
	JoinedAt     time.Time `json:"joined_at" db:"joined_at"`
}
